Crash on joining world (log snippet included)

Joining a new world with a fresh install with Cobblemon 1.5, Counter 1.5, and Unchained 1.5 causes a crash:

[01:25:28] [Server thread/ERROR]: Encountered an unexpected exception
java.lang.NoClassDefFoundError: us/timinc/mc/config/ConfigBuilder
	at us.timinc.mc.cobblemon.chaining.Chaining$Registration.onInit(Chaining.kt:28) ~[cobblemon-unchained-1.5-forge-1.0.0.jar%23561!/:?]
	at us.timinc.mc.cobblemon.chaining.__Registration_onInit_ServerStartedEvent.invoke(.dynamic) ~[cobblemon-unchained-1.5-forge-1.0.0.jar%23561!/:?]
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:73) ~[eventbus-6.0.5.jar%2387!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:315) ~[eventbus-6.0.5.jar%2387!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:296) ~[eventbus-6.0.5.jar%2387!/:?]
	at net.minecraftforge.server.ServerLifecycleHooks.handleServerStarted(ServerLifecycleHooks.java:115) ~[forge-1.20.1-47.2.21-universal.jar%23835!/:?]
	at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:638) ~[client-1.20.1-20230612.114412-srg.jar%23830!/:?]
	at net.minecraft.server.MinecraftServer.m_206580_(MinecraftServer.java:251) ~[client-1.20.1-20230612.114412-srg.jar%23830!/:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.lang.ClassNotFoundException: us.timinc.mc.config.ConfigBuilder
	at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[?:?]
	at cpw.mods.cl.ModuleClassLoader.loadClass(ModuleClassLoader.java:137) ~[securejarhandler-2.1.10.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[?:?]
	at cpw.mods.cl.ModuleClassLoader.loadClass(ModuleClassLoader.java:137) ~[securejarhandler-2.1.10.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[?:?]
	... 9 more

--- a/gradle.properties
+++ b/gradle.properties
@@ -6,5 +6,5 @@ cobblemon_version = 1.5.0+1.20.1
 kff_version = 4.5.0
 maven_group = us.timinc.mc.cobblemon
 archives_base_name = unchained
-mod_version = 1.5-forge-1.0.0
+mod_version = 1.5-forge-1.0.1
 counter_version = 1.5-forge-1.0.1

--- a/gradle.properties
+++ b/gradle.properties
@@ -7,4 +7,4 @@ kff_version = 4.5.0
 maven_group = us.timinc.mc.cobblemon
 archives_base_name = unchained
 mod_version = 1.5-forge-1.0.0
-counter_version = 1.5-forge-1.0.0
+counter_version = 1.5-forge-1.0.1

--- a/src/main/kotlin/us/timinc/mc/cobblemon/chaining/Chaining.kt
+++ b/src/main/kotlin/us/timinc/mc/cobblemon/chaining/Chaining.kt
@@ -4,13 +4,13 @@ import com.cobblemon.mod.common.api.spawning.spawner.PlayerSpawnerFactory
 import net.minecraftforge.event.server.ServerStartedEvent
 import net.minecraftforge.eventbus.api.SubscribeEvent
 import net.minecraftforge.fml.common.Mod
+import us.timinc.mc.cobblemon.chaining.config.ConfigBuilder
 import us.timinc.mc.cobblemon.chaining.config.HiddenBoostConfig
 import us.timinc.mc.cobblemon.chaining.config.IvBoostConfig
 import us.timinc.mc.cobblemon.chaining.config.ShinyBoostConfig
 import us.timinc.mc.cobblemon.chaining.influences.HiddenAbilityBooster
 import us.timinc.mc.cobblemon.chaining.influences.IvBooster
 import us.timinc.mc.cobblemon.chaining.influences.ShinyBooster
-import us.timinc.mc.config.ConfigBuilder
 
 @Mod(Chaining.MOD_ID)
 object Chaining {

--- a/src/main/kotlin/us/timinc/mc/cobblemon/chaining/config/ConfigBuilder.kt
+++ b/src/main/kotlin/us/timinc/mc/cobblemon/chaining/config/ConfigBuilder.kt
@@ -1,0 +1,41 @@
+package us.timinc.mc.cobblemon.chaining.config
+
+import com.google.gson.GsonBuilder
+import java.io.File
+import java.io.FileReader
+import java.io.PrintWriter
+
+class ConfigBuilder<T> private constructor(private val clazz: Class<T>, private val path: String) {
+    companion object {
+        fun <T> load(clazz: Class<T>, path: String): T {
+            return ConfigBuilder(clazz, path)._load()
+        }
+    }
+
+    fun _load(): T {
+        val gson = GsonBuilder()
+            .disableHtmlEscaping()
+            .setPrettyPrinting()
+            .create()
+
+        var config = gson.fromJson("{}" /*default value*/, clazz)
+        val configFile = File("config/$path.json")
+        configFile.parentFile.mkdirs()
+
+        if (configFile.exists()) {
+            try {
+                val fileReader = FileReader(configFile)
+                config = gson.fromJson(fileReader, clazz)
+                fileReader.close()
+            } catch (e: Exception) {
+                println("Error reading config file")
+            }
+        }
+
+        val pw = PrintWriter(configFile)
+        gson.toJson(config, pw)
+        pw.close()
+
+        return config
+    }
+}

--- a/src/main/resources/META-INF/mods.toml
+++ b/src/main/resources/META-INF/mods.toml
@@ -4,7 +4,7 @@ license = "MIT"
 
 [[mods]]
 modId = "unchained"
-version = "1.5-forge-1.0.0"
+version = "1.5-forge-1.0.1"
 displayName = "Cobblemon Unchained"
 description = '''
 Naturally spawning Hidden Abilities. KOs and/or captures increase nearby shiny rates, and can also give perfect IVs to nearby spawns.

